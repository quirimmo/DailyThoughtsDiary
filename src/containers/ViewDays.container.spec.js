import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockFetchItemsThunk = jest
	.fn()
	.mockImplementation(() => ({ type: 'MOCKED_ACTION' }));
jest.mock('../actions/items.action', () => ({
	fetchItemsThunk: () => mockFetchItemsThunk()
}));

jest.mock('moment', () => () => ({
	format: () => '2018–10–28T12:34:56+00:00'
}));

import ViewDaysContainer from './ViewDays.container';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const items = [];
const store = mockStore({ items: [] });
let component;
let instance;

describe('ViewDaysContainer', () => {
	beforeEach(() => {
		const renderer = new ShallowRenderer();
		component = renderer.render(<ViewDaysContainer store={store} />);
		instance = renderer.getRenderOutput();
	});

	it('should be defined', () => {
		expect(ViewDaysContainer).toBeDefined();
	});

	it('should render the component', async () => {
		expect(component).toMatchSnapshot();
	});

	it('should define the fetchItems prop', () => {
		expect(typeof instance.props.fetchItems).toEqual('function');
	});

	it('should define the items prop', () => {
		expect(instance.props.items).toEqual(items);
	});

	it('should dispatch the fetchItemsThunk action', () => {
		instance.props.fetchItems();
		expect(mockFetchItemsThunk).toHaveBeenCalled();
	});
});
