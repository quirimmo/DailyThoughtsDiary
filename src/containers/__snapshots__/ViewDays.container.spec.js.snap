// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewDaysContainer should render the component 1`] = `
<ViewDaysScreen
  fetchItems={[Function]}
  items={Array []}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
/>
`;
